include/alpaka/api/unifiedCudaHip/warp.hpp

```diff
@@ -0,0 +1,429 @@
+/* Copyright 2025 Mehmet Yusufoglu, René Widera
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * Implements CUDA and HIP warp intrinsics via trait specializations.
+ * Exposes backend warp operations through alpaka::onAcc::warp::trait::Op hooks.
+ */
+
+#pragma once
+
+#include "alpaka/api/cuda/Api.hpp"
+#include "alpaka/api/hip/Api.hpp"
+#include "alpaka/core/CudaHipCommon.hpp"
+#include "alpaka/core/common.hpp"
+#include "alpaka/core/config.hpp"
+#include "alpaka/onAcc/internal/warp.hpp"
+
+#include <cstdint>
+#include <type_traits>
+
+namespace alpaka::onAcc::warp::detail
+{
+#if ALPAKA_LANG_CUDA
```

Comment:
Please move the CUDA part into alpaka/api/cuda and the HIP part into alpaka/api/hip